.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,89 @@
+name: 🕵️ Test suite
+
+on:
+  push:
+  pull_request:
+  schedule:
+    # Run every Friday at 23:59 UTC
+    - cron: 59 23 * * 5
+
+jobs:
+  unit-tests:
+    # Run unit tests on different version of python
+    name: 🐍 Python-${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    needs: [ codechecks ]
+    strategy:
+      matrix:
+        python-version: [ pypy2, pypy3, 2.7, 3.6, 3.7, 3.8 ]
```

Comment:
I think you can drop py2 support across the board here, and we'll bump the next release to `1.2` or even `2.0`. Anyone still using py2.7 can freeze their version to `1.1.6`

@RonnyPfannschmidt @psav @quarckster any objection?

Reply:
Yay

Reply:
ahhh I nearly forgot about this PR. going to restructure it as lots of new changes in gh-workflows. Definitely better to drop py2 as per me.